prov/usnic/src/usdf_fabric.c

```diff
@@ -116,13 +186,19 @@ usdf_validate_hints(struct fi_info *hints, struct usd_device_attrs *dap)
 		    fattrp->prov_version != USDF_PROV_VERSION) {
 			return -FI_ENODATA;
 		}
+
+		devn = get_devname(version, dap);
+
 		if (fattrp->name != NULL &&
-                    strcmp(fattrp->name, dap->uda_devname) != 0) {
-			return -FI_ENODATA;
+                    !check_devname(version, dap, NULL, CHECK_DEVNAME_GETINFO)) {
+			USDF_DBG("devname of fabric not matching expected for version (version=%d) (devname='%s')\n", version, devn);
```

Comment:
The `hint requested %s` part is a bit problematic because it is version dependent ... That is why I encapsulated the fetching and the checking of the device name in two functions.
So instated, what do you think of :
`for API version 0x%x, this device's name is %s; device requested in hints does not match`